satellite/rewards/display.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package rewards
+
+// OrganizedOffers contains a list of offers organized by status.
+type OrganizedOffers struct {
+	Active  Offer
+	Default Offer
+	Done    Offers
+}
+
+// OpenSourcePartner contains all data for an Open Source Partner.
+type OpenSourcePartner struct {
+	PartnerInfo
+	PartnerOffers OrganizedOffers
+}
+
+// PartnerSet contains a list of Open Source Partners.
+type PartnerSet []OpenSourcePartner
+
+// OfferSet provides a separation of marketing offers by type.
+type OfferSet struct {
+	ReferralOffers OrganizedOffers
+	FreeCredits    OrganizedOffers
+	PartnerTables  PartnerSet
+}
+
+type referralInfo struct {
+	UserID    string
+	PartnerID string
+}
+
+// OrganizeOffersByStatus organizes offers by OfferStatus.
+func (offers Offers) OrganizeOffersByStatus() OrganizedOffers {
+	var oo OrganizedOffers
+
+	for _, offer := range offers {
+		switch offer.Status {
+		case Active:
+			oo.Active = offer
+		case Default:
+			oo.Default = offer
+		case Done:
+			oo.Done = append(oo.Done, offer)
+		}
+	}
+	return oo
+}
+
+// OrganizeOffersByType organizes offers by OfferType.
+func (offers Offers) OrganizeOffersByType() OfferSet {
+	var (
+		fc, ro, p Offers
+		offerSet  OfferSet
+	)
+
+	for _, offer := range offers {
+		switch offer.Type {
+		case FreeCredit:
+			fc = append(fc, offer)
+		case Referral:
+			ro = append(ro, offer)
+		case Partner:
+			p = append(p, offer)
+		default:
+			continue
+		}
+	}
+
+	offerSet.FreeCredits = fc.OrganizeOffersByStatus()
+	offerSet.ReferralOffers = ro.OrganizeOffersByStatus()
+	offerSet.PartnerTables = organizePartnerData(p)
+	return offerSet
+}
+
+// createPartnerSet generates a PartnerSet from the config file.
+func createPartnerSet() PartnerSet {
+	partners := LoadPartnerInfos()
+	var ps PartnerSet
+	for _, partner := range partners {
+		ps = append(ps, OpenSourcePartner{
+			PartnerInfo: partner,
+		})
+	}
+	return ps
+}
+
+// matchOffersToPartnerSet assigns offers to the partner they belong to.
+func matchOffersToPartnerSet(offers Offers, partnerSet PartnerSet) PartnerSet {
+	for i := range partnerSet {
+		var partnerOffersByName Offers
+
+		for _, o := range offers {
+			if o.Name == partnerSet[i].PartnerInfo.Name {
+				partnerOffersByName = append(partnerOffersByName, o)
+			}
+		}
+
+		partnerSet[i].PartnerOffers = partnerOffersByName.OrganizeOffersByStatus()
+	}
+
+	return partnerSet
+}
+
+// organizePartnerData returns a list of Open Source Partners
+// whose offers have been organized by status, type, and
+// assigned to the correct partner.
+func organizePartnerData(offers Offers) PartnerSet {
+	partnerData := matchOffersToPartnerSet(offers, createPartnerSet())
+	return partnerData
+}
+
+// getTardigradeDomains returns domain names for tardigrade satellites
+func getTardigradeDomains() []string {
```

Comment:
is this the only file where these domain names will need to be accessible? If they will be needed elsewhere, it might make sense to push this function to a higher level so you don't have to worry about multiple sources for this information. Or - is this file the place that makes the most sense for this function in this package?

Reply:
I wanted to get the satellite URL dynamically but didn't find a good easy way to do it so I came up with this temporary solution

Reply:
there might be a way to pass it down from somewhere in `satellite/peer.go`, but I don't know. Maybe that is functionality that can be added in the future.